cmd/print.go

```diff
@@ -13,11 +17,21 @@ var printCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		app := getAppFromContext(cmd.Context())
 		if snippet, ok := app.LookupAndCreatePrintableSnippet(); ok {
-			fmt.Println(snippet)
+			if execCmdOutputFlag != "" {
```

Comment:
This should be moved to the `App` interface. We could refactor the method `LookupAndCreatePrintableSnippet` to `LookupAndPrintSnippet(*string)`. The parameter corresponds to `execCmdOutputFlag`. 

Printing the snippet directly in `print.go` wasn't adhering to the project architecture in the first place.

Reply:
Also, in order to ease testing, `system.WriteFile` should be used instead of `os.WriteFile`.